src/main/java/com/amazon/opendistroforelasticsearch/security/ssl/DefaultOpenDistroSecurityKeyStore.java

```diff
@@ -227,306 +230,324 @@ private void initSSLConfig() {
+                false);
 
-                final String trustedCas = resolve(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMTRUSTEDCAS_FILEPATH,
-                        false);
+            if (httpClientAuthMode == ClientAuth.REQUIRE) {
+                checkPath(trustedCas, SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMTRUSTEDCAS_FILEPATH);
+            }
 
-                if (httpClientAuthMode == ClientAuth.REQUIRE) {
+            final String pemCertFilePath = resolve(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMCERT_FILEPATH, true);
+            final String pemKey = resolve(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_FILEPATH, true);
+
+            try {
+                httpSslContext = buildSSLServerContext(new File(pemKey), new File(pemCertFilePath),
+                    trustedCas == null ? null : new File(trustedCas),
+                    settings.get(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_PASSWORD),
+                    getEnabledSSLCiphers(this.sslHTTPProvider, true), sslHTTPProvider, httpClientAuthMode);
+            } catch (final Exception e) {
+                logExplanation(e);
+                throw new ElasticsearchSecurityException(
+                    "Error while initializing http SSL layer from PEM: " + e.toString(), e);
+            }
 
-                    // if(trustedCas == null ||
-                    // trustedCas.equals(env.config-File().toAbsolutePath().toString())) {
-                    // throw new
-                    // ElasticsearchException(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMTRUSTEDCAS_FILEPATH
-                    // + " must be set if http ssl and client auth is reqested.");
-                    // }
+        } else {
+            throw new ElasticsearchException(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_KEYSTORE_FILEPATH + " or "
+                + SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_FILEPATH
+                + " must be set if http ssl is reqested.");
+        }
+    }
 
-                    checkPath(trustedCas, SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMTRUSTEDCAS_FILEPATH);
+    /**
+     * Cert Issuer DN must be the same
+     * @param newCerts Array of X509Certificates which will replace our current cert
+     * @throws Exception if certificate is invalid
+     */
+    private void validateNewTransportCerts(final X509Certificate[] newCerts) throws Exception {
 
-                }
+        // First time we init certs ignore validity check
+        if (transportCerts == null) {
+            return;
+        }
 
-                final String pemCertFilePath = resolve(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMCERT_FILEPATH, true);
-                final String pemKey = resolve(SSLConfigConstants.OPENDISTRO_SECURITY_SSL_HTTP_PEMKEY_FILEPATH, true);
+        // Check if new X509 certs have been issued by the same Issuer DN
```

Comment:
Trying to understand - we are comparing the set of new certs DNs vs current set DNs? Slightly confused by statement `same Issuer DN`.  Shouldnt it all be the same ? 
* Not familiar with how DNs work.

Reply:
You're right, updated to check for Issuer/Subject and Principal.